File: backend/src/pomelo_orbit/interfaces/api/dto/deployment.py
"""
部署相关 Schema 定义
"""

from datetime import datetime

from pydantic import BaseModel


class DeploymentResp(BaseModel):
    """部署响应"""

    id: str
    application_id: str
    application_name: str | None = None
    operation_type: str
    trigger_type: str
    trigger_ref: str | None
    image_name: str | None
    env_file: str | None
    status: str
    started_at: datetime
    finished_at: datetime | None
    duration_ms: int | None
    error_message: str | None

    model_config = {"from_attributes": True}


class DeploymentDetailResp(DeploymentResp):
    """部署详情响应（包含日志）"""

    log_text: str | None

File: frontend/src/types/cd/deployment.ts
import type { TaskStatus } from '../common';

// 部署记录相关
export interface Deployment {
	id: string
	application_id: string
	application_name: string | null
	operation_type: string
	trigger_type: string
	env_file: string | null
	status: TaskStatus
	started_at: string
	finished_at: string | null
	duration_ms: number | null
	error_message: string | null
}

export interface DeploymentDetail extends Deployment {
	log_text: string | null
}
